File: 2010_2011_tp_note/Ehrenfest2.py
import numpy as np


def Ehrenfest2(X, d):
    """Simulate one step of the lazy Ehrenfest chain."""
    U = np.random.rand()
    if U < d / (d + 1):
        V = np.random.rand()
        X = X + (V > X / d) - (V < X / d)
    return X

File: 2010_2011_tp_note/Ehrenfest2.m
function [X] = Ehrenfest2(X,d )
%UNTITLED6 Simule un pas d'Ehrenfest
%   Detailed explanation goes here
U=rand(1);
if (U<d/(d+1))
    V=rand(1);
    X=X +(V>(X/d))-(V<(X/d));
end
end

